src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -132,6 +132,59 @@ sealed class NumericRange[T](
   // are forgiving: therefore the checks are with the methods.
   private def locationAfterN(n: Int): T = start + (step * fromInt(n))
 
+  private def crossesTheEndAfterN(n: Int): Boolean = {
+    // if we're sure that subtraction in the context of T won't overflow, we use this function
+    // to calculate the length of the range
+    def unsafeRangeLength(r: NumericRange[T]): T = {
+      val diff = num.minus(r.end, r.start)
+      val quotient = num.quot(diff, r.step)
+      val remainder = num.rem(diff, r.step)
+      if (!isInclusive && num.equiv(remainder, num.zero)) quotient else num.plus(quotient, num.one)
+    }
+
+    if (num.equiv(start, end) || n == 0) return n >= 1
+
+    val smallerThanInt = num match {
+      case _: Numeric.ShortIsIntegral | _: Numeric.ByteIsIntegral => true
+      case _                                                      => false
+    }
+
+    val eitherOfBoundariesIsZero = num.equiv(start, zero) || num.equiv(end, zero)
+    val sameSign = num.equiv(num.sign(start), num.sign(end))
+
+    if (eitherOfBoundariesIsZero || sameSign) { // subtraction is safe
+      val len = unsafeRangeLength(this)
+
+      if (smallerThanInt) n >= num.toInt(len) else num.gteq(num.fromInt(n), len)
+    } else {
+      // split to two ranges, which subtraction is safe in both of them (around zero)
+      val closestToZeroFromStart = num.rem(start, step)
```

Comment:
This gives an unsafe range if start = `Int.MinValue` and `step` is a power of two.